src/controllers/recipesController.js

```diff
@@ -0,0 +1,84 @@
+const BaseController = require('./baseController')
+
+class RecipesController extends BaseController {
+  constructor(a) {
+    super(a)
+  }
+
+  async index(req, res) {
+    const recipes = await this.models.recipes.findAll({
+      include: this.models.users,
+    })
+
+    var output = new Set()
+    recipes.forEach((item) => {
+      output.add(item.type)
+    })
+
+    const types = output
+
+    res.render('pages/recipes/index.html.ejs', {
+      recipes,
+      types,
+      context: await this.viewContext(req),
+    })
+  }
+
+  async show(req, res) {
+    const recipe = await this.models.recipes.findOne({
+      where: {
+        id: req.params.id,
+      },
+
+      include: [
+        this.models.users,
+        this.models.recipe_ratings,
+        {
+          model: this.models.comments,
+          include: this.models.users,
+        },
+      ],
+      order: [[this.models.comments, 'createdA', 'DESC']],
+    })
+
+    if (recipe) {
+      res.render('pages/recipes/show.html.ejs', {
+        recipe,
+        totalRating: await recipe.getTotalRating(),
+        myRating: req.user && recipe.recipe_ratings.find((recipeRating) => recipeRating.userId === req.user.id)?.rating,
+        recipesUrl: `${this.utils.fullBaseUrl}/recipes`,
+        context: await this.viewContext(req),
+      })
+    } else {
+      res.sendStatus(404)
+    }
+  }
+
+  async rate(req, res) {
+    const recipe = await this.models.recipes.findOne({ where: { id: req.params.id } })
```

Comment:
Galima naudoti `findByPk` (https://sequelize.org/docs/v6/core-concepts/model-querying-finders/#findbypk):
```suggestion
    const recipe = await this.models.recipes.findByPk(req.params.id)
```